src/Importer/CustomerOptionPriceCsvImporter.php

```diff
@@ -0,0 +1,221 @@
+    ];
+
+    /** @var EntityManagerInterface */
+    protected $entityManager;
+
+    /** @var CustomerOptionPriceUpdaterInterface */
+    protected $priceUpdater;
+
+    /** @var SenderInterface */
+    protected $sender;
+
+    /** @var TokenStorageInterface */
+    protected $tokenStorage;
+
+    /** @var ProductRepositoryInterface */
+    protected $productRepository;
+
+    /** @var ProductInterface[] */
+    protected $products = [];
+
+    public function __construct(
+        CustomerOptionPriceUpdaterInterface $priceUpdater,
+        EntityManagerInterface $entityManager,
+        SenderInterface $sender,
+        TokenStorageInterface $tokenStorage,
+        ProductRepositoryInterface $productRepository
+    ) {
+        $this->priceUpdater      = $priceUpdater;
+        $this->entityManager     = $entityManager;
+        $this->sender            = $sender;
+        $this->tokenStorage      = $tokenStorage;
+        $this->productRepository = $productRepository;
+    }
+
+    /** {@inheritdoc} */
+    public function import(string $source): array
+    {
+        $csv = $this->readCsv($source);
+
+        // Handle updates
+        $i      = 0;
+        $failed = [];
+        foreach ($csv as $lineNumber => $data) {
+            if (!$this->isRowValid($data)) {
+                $failed[$lineNumber] = ['data' => $data, 'message' => 'Data is invalid'];
+
+                continue;
+            }
+
+            try {
+                $product = $this->getProduct($data['product_code']);
+                Assert::isInstanceOf(
+                    $product,
+                    ProductInterface::class,
+                    sprintf('Product with code "%s" not found', $data['product_code'])
+                );
+
+                $price = $this->priceUpdater->updateForProduct(
+                    $data['customer_option_code'],
+                    $data['customer_option_value_code'],
+                    $data['channel_code'],
+                    $product,
+                    $data['valid_from'],
+                    $data['valid_to'],
+                    $data['type'],
+                    (int) $data['amount'],
+                    (float) $data['percent']
+                );
+
+                // Add the value price to the product so we can use it in later validations.
+                $product->addCustomerOptionValuePrice($price);
+
+                $this->entityManager->persist($price);
+
+                if (++$i % self::BATCH_SIZE === 0) {
+                    $this->entityManager->flush();
+                }
+            } catch (\Throwable $exception) {
+                $failed[$lineNumber] = ['data' => $data, 'message' => $exception->getMessage()];
+            }
+        }
+
+        $this->entityManager->flush();
+
+        $this->sendFailReport($failed);
+
+        return ['imported' => $i, 'failed' => count($failed)];
+    }
+
+    private function sendFailReport(array $failed): void
```

Comment:
This should definitely be customizable with it's own service. (Because people might want to override it) Maybe we can get around by using a custom logger and handler in our case.